packages/headless/src/controllers/core/context/headless-core-context.ts

```diff
@@ -66,6 +85,17 @@ export function buildCoreContext(engine: CoreEngine): Context {
   const {dispatch} = engine;
   const getState = () => engine.state;
 
+  const initialState = validateInitialState(
```

Comment:
We should add some tests for this behaviour in `headless-core-context-test`: That is: checking that creating the controller with an initial state correctly dispatch the `setContext `action